src/messages/direct.rs

```diff
@@ -54,29 +58,71 @@ pub enum DirectMessage {
 #[derive(Eq, PartialEq, Serialize, Deserialize, Debug, Hash)]
 pub enum BootstrapResponse {
     /// This response means that the new peer is clear to join the section. The connection infos of
-    /// the Elders of the section are provided.
-    Join(Vec<ConnectionInfo>),
+    /// the Elders of the section and the section prefix are provided.
+    Join {
+        prefix: Prefix<XorName>,
+        conn_infos: Vec<ConnectionInfo>,
+    },
     /// The new peer should retry bootstrapping with another section. The set of connection infos
     /// of the members of that section is provided.
     Rebootstrap(Vec<ConnectionInfo>),
     /// An error has occurred
     Error(BootstrapResponseError),
 }
 
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Hash)]
+pub struct RelocatePayload {
+    pub details: SignedRelocateDetails,
+    /// The new id (`PublicId`) of the node signed using its old id, to prove the node identity.
+    pub signature_of_new_id_with_old_id: Signature,
```

Comment:
No, but it would fall into the same prefix. It's not computationally feasible to generate a `FullId` that would have a given name. That's why the node needs to first send `BootstrapRequest` with the name, then receives `BootstrapResponse` which contains a `Prefix` that matches that name and only then can it generate new id that matches the prefix too (which is easy, but gets harder the more bits the prefix has).